rca/navigation/models.py

```diff
@@ -83,6 +83,8 @@ def get_api_representation(self, value, context=None):
 
         if value["page"] and not value["url"]:
             value["url"] = Page.objects.get(id=value["page"]).url
+            if not value["title"]:
+                value["title"] = Page.objects.get(id=value["page"]).title
```

Comment:
If this code path is hit then the page query is run again, so might be best to store the result of the previous query in a variable.
Could it ever be the case that the page referred to in `value["page"]` will have been removed but this reference not updated? If so the `.get()` would raise an exception that could break the page

Reply:
I've added a catch for this... but it's a bit moot, as the resulting data is cached on the legacy site so stale page IDs could well end up in there anyway.